docs/sources/flow/reference/components/loki.source.azure_event_hubs.md

```diff
@@ -0,0 +1,129 @@
+ `disallow_custom_messages`  | `bool`               | Whether to ignore messages that don't match the [schema](https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/resource-logs-schema) for Azure resource logs | `false`                          | no       
+ `relabel_rules`             | `RelabelRules`       | Relabeling rules to apply on log entries.                                                                                                                               | `{}`                             | no       
+
+`assignor` values can be either `"range"`, `"roundrobin"`, or `"sticky"`.
+
+The `relabel_rules` field can make use of the `rules` export value from a
+`loki.relabel` component to apply one or more relabeling rules to log entries
+before they're forwarded to the list of receivers in `forward_to`.
+
+## Blocks
+
+The following blocks are supported inside the definition of `loki.source.azure_event_hubs`:
+
+ Hierarchy      | Name             | Description                                        | Required 
+----------------|------------------|----------------------------------------------------|----------
+ authentication | [authentication] | Authentication configuration with Azure Event Hub. | yes      
+
+[authentication]: #authentication-block
+
+### authentication block
+
+The `authentication` block defines the authentication method when communicating with Azure Event Hub.
+
+ Name                | Type           | Description                                                               | Default | Required 
+---------------------|----------------|---------------------------------------------------------------------------|---------|----------
+ `mechanism`         | `string`       | Authentication mechanism.                                                 |         | yes      
+ `connection_string` | `string`       | Event Hubs ConnectionString for authentication on Azure Cloud.            |         | no       
+ `scopes`            | `list(string)` | Access token scopes. Default is `fully_qualified_namespace` without port. |         | no       
+
+`mechanism` supports the values `"connection_string"` and `"oauth"`. If `"connection_string"` is used,
+you must set the `connection_string` attribute. If `"oauth"` is used, you must configure one of the supported credential
+types as documented
+here: https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azidentity/README.md#credential-types via environment
+variables or Azure CLI.
+
+## Labels
+
+The `labels` map is applied to every message that the component reads.
+
+The following internal labels prefixed with `__` are available but are discarded if not relabeled:
+
+- `__meta_kafka_message_key`
+- `__meta_kafka_topic`
+- `__meta_kafka_partition`
+- `__meta_kafka_member_id`
+- `__meta_kafka_group_id`
+- `__azure_event_hubs_category`
```

Comment:
This should be documented in the Arguments header rather than being given its own header section; #3145 details some of the recommendations here.

Reply:
Okay, I'll fix that! FYI I got this from here: https://grafana.com/docs/agent/latest/flow/reference/components/loki.source.kafka/#labels

Reply:
Thanks for the pointer! I updated the other doc page over in #3448.